### PR TITLE
feat(typescript): introduce

### PR DESCRIPTION
**Pull request by AugustinMauroy**
## Proposed changes

[//]: # (Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.)

introduce typescript on the docs.

[#10117](https://github.com/webdriverio/webdriverio/issues/10117)

## Types of changes

[//]: # (What types of changes does your code introduce to WebdriverIO?)
[//]: # (_Put an `x` in the boxes that apply_)

- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [X] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/main/CONTRIBUTING.md) doc
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] I have added proper type definitions for new commands (if appropriate)

## Further comments

[//]: # (If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...)

### CODE REVIEW

File: website/community/Team.md [hunk 1]
```diff
@@ -3,7 +3,7 @@ id: team
 title: Team
 ---
 
-import Card from '@site/src/theme/card.jsx';
+import Card from '@site/src/theme/card.tsx';
```
Comment (christian-bromann) [inline website/community/Team.md:6]: Can we add this to https://github.com/webdriverio/webdriverio/blob/main/website/src/theme/MDXComponents.js instead?

File: website/package.json [hunk 2]
```diff
@@ -41,5 +41,10 @@
       "last 1 firefox version",
       "last 1 safari version"
     ]
+  },
+  "devDependencies": {
+    "@docusaurus/module-type-aliases": "^2.4.1",
+    "@tsconfig/docusaurus": "^2.0.0",
+    "typescript": "^5.1.6"
```
Comment (christian-bromann) [inline website/package.json:48]: I wonder if we can just move them all into the `dependency` section. I think it wouldn't make any differences.

File: website/src/pages/index.tsx [hunk 3]
```diff
@@ -6,15 +6,15 @@ import Link from '@docusaurus/Link'
...
+import LogoCarousel from './components/LogoCarousel'
+import Features from './components/Features'
+import Section from './components/Section'
+import Highlight from './components/Highlight'
+import Robot from './components/Robot'
+import CreateProjectAnimation from './components/CreateProjectAnimation'
 
 import styles from './styles.module.css'
-import { logos, features, LHIntregrationExample, SetupExample, ComponentTestingExample } from '../constants.js'
+import { logos, features, LHIntregrationExample, SetupExample, ComponentTestingExample } from '../constants'
```
Comment (christian-bromann) [inline website/src/pages/index.tsx:17]: Let's keep all `.js` endings as it is a required thing in ESM and Docusaurus might support it in the future.

**Review by christian-bromann — COMMENTED** [review]
This is great progress 👍  let me know if there is anything I can help you with!

File: website/community/Team.md (hunk 1 above)
Comment (AugustinMauroy) [reply website/community/Team.md:6]: I will test be good idea !

File: website/package.json (hunk 2 above)
Comment (AugustinMauroy) [reply website/package.json:48]: it's from `npm install --save-dev typescript @docusaurus/module-type-aliases @tsconfig/docusaurus`
Ref: [TypeScript Support](https://docusaurus.io/docs/typescript-support)

Comment (christian-bromann) [reply website/package.json:48]: Let's keep only `dependencies` or `devDependencies`. Having both might confuse folks.

**Comment by christian-bromann** [comment]
While you at it, can you please move out the `website/src/pages/components` directory out of the `website/src/pages` folder and update all imports?

**Comment by christian-bromann** [comment]
@AugustinMauroy are you still planning to continue working on this?

**Comment by AugustinMauroy** [comment]
oops I completely skipped this pr. I will comme back during this week

**Comment by AugustinMauroy** [comment]
Hey sorry I didn't manage to finish this pr.

**Pull request closed without merging**